device/transport/amqp/devdoc/device_amqp_requirements.md

```diff
@@ -122,9 +136,9 @@ This method is deprecated. The `AmqpReceiver` object and pattern is going away a
 
 **SRS_NODE_DEVICE_AMQP_16_015: [**The `updateSharedAccessSignature` method shall save the new shared access signature given as a parameter to its configuration.**]**
 
-**SRS_NODE_DEVICE_AMQP_06_010: [** The `updateSharedAccessSignature` method shall call the amqp transport `putToken` method with the first parameter audience, created from the sr of the sas signature, the next parameter of the actual sas, and a callback. *]**
+**SRS_NODE_DEVICE_AMQP_06_010: [** If the AMQP connection is established, the `updateSharedAccessSignature` method shall call the amqp transport `putToken` method with the first parameter `audience`, created from the `sr` of the shared access signature, the actual shared access signature, and a callback. **]**
 
-**SRS_NODE_DEVICE_AMQP_06_011: [** The `updateSharedAccessSignature` method shall call the `done` callback with a null error object and a SharedAccessSignatureUpdated object as a result, indicating the client does NOT need to reestablish the transport connection. **]**
+**SRS_NODE_DEVICE_AMQP_06_011: [** The `updateSharedAccessSignature` method shall call the `done` callback with a `null` error object and a `SharedAccessSignatureUpdated` object as a result, indicating the client does NOT need to reestablish the transport connection. **]**
```

Comment:
>indicating the client does NOT need to reestablish the transport connection. [](start = 191, length = 76)

I have a few questions/issues with this:
1) What is it specifically that indicates that the client doesn't need to re-establish the transport connection?  It sounds like there is an inference here that could be made explicit (e.g. add a reconectionNeeded parameter)
2) There's some religion that I'm missing here.  Why does the client need to re-establish the connection?  If it's because the device transport needs to re-establish links after reconnecting, why can't it happen automatically at this level?

Reply:
historically the client would take care of reconnecting the transport so the transport needed to communicate the need to reconnect and reestablish links. now that we're pushing this logic down in the transport itself, the client itself doesn't need to take care of reconnecting anything.
I've updated the requirement to be explicit about the needToReconnect value.

---
In reply to: [140955441](https://github.com/Azure/azure-iot-sdk-node/pull/89#discussion_r140955441) [](ancestors = 140955441)